homeassistant/util/__init__.py

```diff
@@ -233,35 +233,42 @@ def __init__(self, min_time, limit_no_throttle=None):
         self.limit_no_throttle = limit_no_throttle
 
     def __call__(self, method):
-        lock = threading.Lock()
-
         if self.limit_no_throttle is not None:
             method = Throttle(self.limit_no_throttle)(method)
 
+        # We want to be able to differentiate between function and method calls
+        # All methods have the classname in their qualname seperated by a '.'
+        # Functions have a '.' in their qualname if defined inline, but will
+        # be prefixed by '.<locals>.' so we strip that out.
+        is_func = '.' not in method.__qualname__.split('.<locals>.')[-1]
```

Comment:
@SEJeff do you know if there is a better way to distinguish if we are decorating a function or a method?

The problem is that the moment the decorator is called, it is a function (unbound class method). When we get into `wrapper`, we are unable to distinguish because it is our wrapper method that got bound, which we can't inspect from inside the method.